src/ja/nicomanga/src/eu/kanade/tachiyomi/extension/ja/nicomanga/Nicomanga.kt

```diff
@@ -0,0 +1,156 @@
+package eu.kanade.tachiyomi.extension.ja.nicomanga
+
+import android.app.Application
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.injectLazy
+import java.net.URL
+
+class Nicomanga : HttpSource() {
+    override val baseUrl: String = "https://nicomanga.com"
+    override val lang: String = "ja"
+    override val name: String = "Nicomanga"
+    override val supportsLatest: Boolean = true
+    private val application: Application by injectLazy()
+    private val thumbnailURLRegex: Regex = "background-image:[^;]url\\s*\\(\\s*'([^?']+)".toRegex()
+    private val statusRegex: Regex = "(?<=-)[^.]+".toRegex()
+    private val urlRegex: Regex = "(?<=manga-)[^/]+(?=\\.html\$)".toRegex()
+    private val floatRegex: Regex = "\\d+(?:\\.\\d+)?".toRegex()
+    private val chapterIdRegex: Regex = "(?<=imgsListchap\\()\\d+".toRegex()
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder().addInterceptor { chain ->
+        val request = chain.request()
+        val headers = request.headers.newBuilder()
+        if ((request.headers["Referer"] ?: "") != "") {
+            headers.add("Referer", baseUrl)
+        }
+        chain.proceed(request.newBuilder().headers(headers.build()).build())
+    }.build()
+
+    private fun mangaListParse(response: Response): MangasPage {
+        val doc = Jsoup.parse(response.body.string())
+        val mangaList: ArrayList<Element> = doc.select(".row > .thumb-item-flow")
+        val hasNextPage =
```

Comment:
why not use `ParsedHttpSource`?
You can use something like `.pagination > li:last-child > a:not(.disabled)` for `*MangaNextPageSelector`